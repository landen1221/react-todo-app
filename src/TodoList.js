import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import { v4 as uuid } from "uuid";
import Todo from "./Todo";

const TodoList = () => {
    const TEST_TODO = [{key: uuid(), todo: 'Walk the dog'}]
    const TEST_COMPLETED = [{key: '', todo: ''}]

  const [tasks, setTask] = useState(TEST_TODO);
  
  const [completedTasks, setCompletedTask] = useState(TEST_COMPLETED);

  const addTask = ({ todo }) => {
    setTask((tasks) => [...tasks, { todo, key: uuid() }]);
  };


  const removeTask = (key) => {
    setCompletedTask((completedTasks) => [...completedTasks, tasks.filter(obj => obj.key === key)[0]]); 
    console.log(completedTasks) 
    setTask((tasks) => tasks.filter(obj => obj.key !== key))
  }



  return (
    <div>
      <NewTodoForm addTask={addTask} />
      <ul>
          {tasks.map (({todo, key}) => (
              <Todo key={key} id={key} todo={todo} removeTask={removeTask} />
          ))}
      </ul>
      <br/>
      <hr/>
      <br/>
      <h3>Completed</h3>
      <ul>
          
          {completedTasks.map (({todo, key}) => (
              <Todo key={key} id={key} todo={todo} removeTask={removeTask} />
          ))}
      </ul>

    </div>
  );
};

export default TodoList;
