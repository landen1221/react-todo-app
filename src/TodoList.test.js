import { render } from '@testing-library/react';
import TodoList from './TodoList'

test('renders learn react link', () => {
    render(<TodoList />);
});
  
it('should match snapshot', () => {
    const {asFragment} = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
})