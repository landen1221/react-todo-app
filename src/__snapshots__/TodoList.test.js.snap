// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h3>
        This is the form to add a task!
      </h3>
      <form>
        <label
          for="todo"
        >
          New Todo: 
        </label>
        <input
          id="todo"
          name="todo"
          placeholder="ie: Walk the dog"
          type="text"
          value=""
        />
        <button>
          Add Task
        </button>
      </form>
    </div>
    <ul>
      <li
        class="Todo"
        id="8bc8fe7b-3936-49ef-96b7-04cf2f951530"
      >
        Walk the dog  
        <a
          class="Todo-close"
          href="#"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.47 250.9C88.82 328.1 158 397.6 224.5 485.5c72.3-143.8 146.3-288.1 268.4-444.37L460 26.06C356.9 135.4 276.8 238.9 207.2 361.9c-48.4-43.6-126.62-105.3-174.38-137z"
            />
          </svg>
        </a>
      </li>
    </ul>
    <br />
    <hr />
    <br />
    <h3>
      Completed
    </h3>
    <ul>
      <li
        class="Todo"
        id=""
      >
          
        <a
          class="Todo-close"
          href="#"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.47 250.9C88.82 328.1 158 397.6 224.5 485.5c72.3-143.8 146.3-288.1 268.4-444.37L460 26.06C356.9 135.4 276.8 238.9 207.2 361.9c-48.4-43.6-126.62-105.3-174.38-137z"
            />
          </svg>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
